### DISCUSSION TITLE
23

### DISCUSSION THREAD

**Discussion by jojoldu** [jojoldu/blog-comments#738] [Comments]
http://jojoldu.tistory.com/23

**Comment by jojoldu** [comment]
작성자: 지나가는 개발자

grunt.js의 namespace나 processName이 어떠한 방식으로 돌아가고 의미가 어떤지가 궁금합니다

**Comment by jojoldu** [comment]
작성자: 창천향로

안녕하세요! 방문해주셔서 감사합니다!
namespace는 Handlebars.templates 라는 이름으로 compile 함수를 실행시키겠다는 의미입니다.

processName은 .handlebars 파일들을 하나하나 컴파일하면서 해당 파일들을 어떤 이름의 function으로 template.js에 등록할건지를 결정하는것입니다.

크롬 개발자도구에서 아래와 같이 명령어를 입력해보시면 정규표현식의 결과가 어떻게 되는지 확인하실수 있습니다^^

var pattern=/handlebar\/(.+\/)*(.+)\.handlebars/gi;
var a = pattern.exec("/handlebar/basic.handlebars")[2];
console.log(a);

**Comment by jojoldu** [comment]
작성자: 지나가는 개발자2

레거시 스펠체크를 ..

**Comment by jojoldu** [comment]
작성자: 창천향로 창천향로

으앗 ㅋㅋㅋㅋㅋ
감사합니다!!! ㅠㅠ 바로 수정하였습니다!

**Comment by jojoldu** [comment]
작성자: Kiukjung

실무 프로젝트에서 handlebars 를 사용할까 고민중입니다.
Spring mvc에서 사용할려구 합니다.
Jsp에 비해서 어떠한 장점이 있을까요? Handlebars 러닝커가 높은 언어인가요?

**Comment by jojoldu** [comment]
작성자: 창천향로 창천향로

방문해주셔서 감사합니다^^
JSP와 여기서 소개된 Handlebars.js는 용도와 역할이 서로 다릅니다.
만약 Handlebars.java를 사용하신다면 JSP와 비교가 될 수 있겠지만
여기서 소개하는 Handlebars.js는 Javascript 라이브러리로 랜더링이 끝난뒤 (즉, HTML Dom이 다 그려진 뒤)에 서버 통신 없이 화면 변경이 필요할 경우를 위한것입니다. 

둘의 용도가 달라 비교할수는 없을것 같습니다^^; 
러닝커브는 어떻게 사용하느냐에 따라 다르지만 높은 편은 아니라고 생각합니다. 결국 화면을 그리는것이 전부인 라이브러리라서요^^

**Comment by jojoldu** [comment]
작성자: ssosso

안녕하세요^^ 새로 만드는 프로젝트에 Handlebars.js 적용중인데 많은 도움이 되었습니다.
앞으로 좋은 포스팅 많이 부탁드립니다^^

**Comment by jojoldu** [comment]
작성자: 창천향로 창천향로

도움이 되셨다니 다행입니다^^;
더 좋은 포스팅 올리도록 노력하겠습니다.
감사합니다!

**Comment by sgpark94** [comment]
안녕하세요 handlebar.js 설명글 너무 잘읽어보았습니다!
저는 퍼블리셔인데 handlebars.js로 퍼블리싱만 진행하려고 하는데,
위에서 말씀해주신 방법은 gradle, grunt로 배포까지하는걸 알려주셔서 어떻게 적용을 하면 좋을지 잘모르겠더라구요 ㅜㅜ 이 부분에 대하여 질문드려도 될까요?
